xsdvalidate/xsd_validate_activity.go

```diff
@@ -0,0 +1,226 @@
+package xsdvalidate
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"io/fs"
+	"net/http"
+	"net/url"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"regexp"
+	"strings"
+	"time"
+
+	"go.artefactual.dev/tools/temporal"
+)
+
+const XSDValidateActivityName = "xsd-validate-activity"
+
+type XSDValidateActivity struct{}
+
+func NewXSDValidateActivity() *XSDValidateActivity {
+	return &XSDValidateActivity{}
+}
+
+type XSDValidateActivityParams struct {
+	// DirectoryPath is the path of the files to be validated.
+	DirectoryPath string
+
+	// Pattern is the regular expression for files to validate.
+	Pattern string
+
+	// XSDFilePathOrURL is the path of the XSD file to use for validation.
+	XSDFilePathOrURL string
+}
+
+type XSDValidateActivityResult struct {
+	Failed []string
+}
+
+// Execute checks all files, in a givenrror parsing regexp directory, matching a certain pattern.
```

Comment:
Another thought, we'll need to improve the readme and docs for this project, but at least we should mention the `xmllint` requirement in this comment.

Reply:
Good call... I'll add something to the README.

Reply:
(Added.)